Reaqtive/Core/Tests.Reaqtive.Linq/Operators/ElementAt.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT License.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+using Reaqtive;
+using Reaqtive.Testing;
+using Reaqtive.TestingFramework;
+
+namespace Test.Reaqtive.Operators
+{
+    [TestClass]
+    public partial class ElementAt : OperatorTestBase
+    {
+        [TestInitialize]
+        public void Initialize()
+        {
+            base.TestInitialize();
+        }
+
+        [TestCleanup]
+        public void Cleanup()
+        {
+            base.TestCleanup();
+        }
+
+        #region ElementAt
+
+        [TestMethod]
+        public void ElementAt_ArgumentChecking()
+        {
+            ReactiveAssert.Throws<ArgumentNullException>(() => Subscribable.ElementAt(default(ISubscribable<int>), 0));
+            ReactiveAssert.Throws<ArgumentOutOfRangeException>(() => Subscribable.ElementAt(DummySubscribable<int>.Instance, -1));
+        }
+
+        [TestMethod]
+        public void ElementAt_SaveAndReload()
+        {
+            var state = Scheduler.CreateStateContainer();
+
+            var checkpoints = new[] {
+                OnSave(240, state),
+                OnLoad(320, state),
+            };
+
+            var xs = Scheduler.CreateHotObservable(
+                OnNext(230, 1),
+                // OnSave
+                OnNext(280, 0),
+                OnNext(290, 1),
+                OnNext(310, 0),
+                // OnLoad
+                OnNext(330, 1),
+                OnNext(340, 0),
+                OnNext(350, 1),
+                OnNext(360, 0),
+                OnNext(370, 1),
+                OnNext(380, 0),
+                OnNext(390, 1),
+                OnCompleted<int>(400)
+            );
+
+            var res = Scheduler.Start(() =>
+                xs.ElementAt(4).Apply(Scheduler, checkpoints)
```

Comment:
Not a review comment per se, but something I'd like to point out for folks watching this PR and wondering why this test emits `360, 0` as the 4th element, rather than `330, 1`.

* At the level of Reaqtive, we don't have replay of missed events (in here, the time range `280..310`), so when the operator recovers at time `320`, it goes back in time to the point where only `1` element was received.
* At the level of Reaqtor, inputs to a query engine would have sequence IDs, and upon recovery, events would get replayed. This is what the `IReliable*` interfaces deal with.